packages/lesswrong/lib/react-headroom/index.tsx

```diff
@@ -124,10 +148,11 @@ export default class Headroom extends Component {
   }
 
   getScrollY = () => {
-    if (this.props.parent().pageYOffset !== undefined) {
-      return this.props.parent().pageYOffset
-    } else if (this.props.parent().scrollTop !== undefined) {
-      return this.props.parent().scrollTop
+    const parent = this.props.parent()
+    if ('pageYOffset' in parent && parent.pageYOffset !== undefined) {
+      return parent.pageYOffset
+    } else if ('scrollTop' in parent && parent.scrollTop !== undefined) {
+      return parent.scrollTop
```

Comment:
This would've been more annoying to type with `AnyBecauseTodo` than just making the language perform the correct type narrowing itself.  It should be functionally equivalent to the previous code (there'll never be a case where  `parent.pageYOffset !== undefined` returns `true` but `pageYOffset in parent` returns false).